src/utils.js

```diff
@@ -4,3 +4,11 @@
 // FIXME: Could not find a way to define a Callback<T> type with multiple
 //        signatures.
 export type Callback = (?Error, any) => void;
+
+export function composeAsync (f1: (...args: Array<*>) => Promise<*>,
+                       f2: (*) => Promise<*>):
+                      (...args: Array<*>) => Promise<*> {
+  return async function composed (...args: Array<*>): Promise<*> {
+    return f2(await f1(...args))
```

Comment:
The more I think about it, the more I believe either sync exceptions are bugs, or they should be handled in the component function who choosed to mix sync and async calls possibly throwing domain exceptions, not in the code calling the composed function.

Would you mind waiting to actually get the issue before handling this case?